ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -776,6 +776,18 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			return err
 		}
 
+		// Delete all workers if worker group is suspended and skip reconcile
+		if worker.Suspend != nil && *worker.Suspend {
+			if _, err := r.deleteAllPods(ctx, common.RayClusterGroupPodsAssociationOptions(instance, worker.GroupName)); err != nil {
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, string(utils.FailedToDeleteWorkerPod),
```

Comment:
Minor: I am considering whether we should create a new event type instead of reusing `FailedToDeleteWorkerPod`. The reason is that `Delete` relies on the verb `delete` in Role or ClusterRole, whereas `DeleteAllOf` relies on the verb `deletecollection`. Any thoughts? cc @rueian @andrewsykim 


We are not necessary to fix this in this PR, and we can leave it as a good first issue for newbie contributors instead.

Reply:
New event type makes sense to me, I can fix it here or we can keep it separate

Reply:
we can leave it as a good first issue.

Reply:
Maybe a `FailedToDeleteWorkerPodCollection` type?

Reply:
open an issue: https://github.com/ray-project/kuberay/issues/2672